src/fmtcore.jl

```diff
@@ -218,6 +219,36 @@ function _pfmt_floate(out::IO, sch::Char, zs::Integer, u::Real, prec::Int, e::In
 end
 
 
+# Pull in definition of signif from v0.6.2 base, since it is currently broken for
+# BigFloat and DecFP (at least) on master
+
+@static if VERSION >= v"0.7.0-DEV.4804"
+# adapted from Matlab File Exchange roundsd: http://www.mathworks.com/matlabcentral/fileexchange/26212
```

Comment:
_Do not use the Matlab file exchange_.  While the original author is required to license their work as BSD, by downloading it from the file exchange you're agreeing to this additional clause in their [terms of use](https://www.mathworks.com/matlabcentral/termsofuse.html):

> Content submitted to File Exchange may only be used with MathWorks products.

Besides that, a URL in source code does not satisfy the terms of the BSD license for the original author — that copyright notice should be added or linked to in the LICENSE.

Reply:
That was taken directly from Julia base/floatfunctions.jl (in v0.6.x).
If there is a problem with Julia base, then it should be corrected there as well.

Reply:
As soon as it is fixed in base (and hopefully supported by Compat.jl), this can be changed to simply call the new form, i.e. `round(ax; sigdig=fs.prec + 1)`.

Reply:
Yep - good advice (to stay away from MathWorks).
I hadn't thought about it, since I'd assumed that things in Base had been well vetted.
(and you know what happens when you *assume* anything!)

Reply:
I found the original change - and it's an ancient one! 5.5 years old, from Sept. 4, 2012 (https://github.com/JuliaLang/julia/commit/4c88c229955be98b79c2d084e105434f0577b936) by HarlanH, committed by Stefan.
Not sure if the violation is considered to be by the person who made the PR, or the person who actually placed it into the Julia source code against the ToU.